playwright/_impl/_connection.py

```diff
@@ -337,6 +354,40 @@ def _replace_guids_with_channels(self, payload: Any) -> Any:
             return result
         return payload
 
+    async def wrap_api_call(
+        self, cb: Callable[[], Coroutine], is_internal: bool = False
+    ) -> Any:
+        zones = self._api_zone.get()
+        if zones[0] is not None:
+            result = cb()
+            return (
+                await cast(Coroutine, result) if asyncio.iscoroutine(result) else result
+            )
+        task = asyncio.current_task(self._loop)
+        st: List[inspect.FrameInfo] = getattr(task, "__pw_stack__", inspect.stack())
```

Comment:
This assumes there were no async hops on the way from API call to wrap_api_call. I believe this is the case, but I'm not sure we should rely upon in. Consider leaving _async wrapper in place as a safer option.